src/odemis/acq/stitching/test/tiledacq_test.py

```diff
@@ -311,6 +311,119 @@ def test_whole_procedure(self):
         self.assertIsInstance(data[0], model.DataArray)
         self.assertEqual(len(data[0].shape), 2)
 
+    def test_refocus(self):
+        """Test the refocus function which provides the z levels for the zstack."""
+        area = (-0.001, -0.001, 0.001, 0.001)
+        overlap = 0.2
+        focus_points = [[-0.00019889999999999998, -0.0001485167330631956, 2.5185183e-05],
+                        [0.0, -0.0001485167330631956, -1.8148142e-05],
+                        [0.00019889999999999998, -0.0001485167330631956, -3.382517e-06],
+                        [-0.00019889999999999998, 0.0, 1.3804983e-05],
+                        [0.0, 0.0, 2.320607e-06], [0.00019889999999999998, 0.0, 3e-05],
+                        [-0.00019889999999999998, 0.0001485167330631956, -1.78125e-05],
+                        [0.0, 0.0001485167330631956, -8.124999e-06],
+                        [0.00019889999999999998, 0.0001485167330631956, 4.765627e-06]]
+        focus_range = self.focus.getMetadata().get(model.MD_POS_ACTIVE_RANGE)
+        axis_range = self.focus.axes['z'].range
+
+        # Test 1 when z level is not provided
+        tiled_acq_task = TiledAcquisitionTask(self.fm_streams, self.stage,
+                                              area=area, overlap=overlap, future=model.InstantaneousFuture(),
+                                              focus_points=focus_points)
+        # TODO need the following metadata in mimas-sim and meteor-sim
```

Comment:
I think it would be nice to just add this in this PR. I'm pretty sure the TODO will get lost here :)

Reply:
In mimas and enzel, the metadata of POS_ACTIVE_RANGE can be added, in meteor SAFE_REL_RANGE can be added in respective microscope files. @pieleric should I change the three microscope files or leave them as it is. Irrespective of adding the metadata in the microscope file, I will make sure no error arises when the metadata is not defined.